File: lingting-polaris-grpc/src/main/java/live/lingting/polaris/grpc/util/JvmHookHelper.java

package live.lingting.polaris.grpc.util;

import lombok.experimental.UtilityClass;

/**
 * @author lixiaoshuang
 */
@UtilityClass
public class JvmHookHelper {

	/**
	 * Add JVM callback hooks.
	 * @param runnable Functional interface
	 */
	public static boolean addShutdownHook(Runnable runnable) {
		Runtime.getRuntime().addShutdownHook(new Thread(runnable));
		return true;
	}

}

File: lingting-polaris-grpc/src/main/kotlin/live/lingting/polaris/grpc/util/JvmHookHelper.kt
package live.lingting.polaris.grpc.util

/**
 * @author lixiaoshuang
 */
class JvmHookHelper private constructor() {
    init {
        throw UnsupportedOperationException("This is a utility class and cannot be instantiated")
    }

    companion object {
        /**
         * Add JVM callback hooks.
         *
         * @param runnable Functional interface
         */
        fun addShutdownHook(runnable: Runnable?): Boolean {
            Runtime.getRuntime().addShutdownHook(Thread(runnable))
            return true
        }
    }
}
